classes/ObjectModel.php

```diff
@@ -385,12 +391,15 @@ public function getFieldsLang()
      */
     public function getAssociatedLanguage(): Language
     {
-        $language = new Language($this->id_lang);
-        if (null === $language->id) {
-            $language = new Language((int) Configuration::get('PS_LANG_DEFAULT'));
+        if (null !== $this->lang_associated) {
+            return $this->lang_associated;
+        }
+        $this->lang_associated = new Language($this->id_lang);
+        if (null === $this->lang_associated->id) {
```

Comment:
```suggestion
        if (!Validate::isLoadedObject($this->lang_associated)) {
```

Reply:
Just a suggestion as I often see this kind of code to check an object is correctly loaded, but not a mandatory modif 😉